waspc/data/Generator/templates/server/src/routes/auth/passport/google/googleDefaults.js

```diff
@@ -0,0 +1,35 @@
+import { upsertUserWithRandomPassword } from '../../../../core/auth.js'
+
+// Default implementation if there is no `auth.methods.google.configFn`.
+export function configFn() {
+  const clientId = process.env['GOOGLE_CLIENT_ID']
+  const clientSecret = process.env['GOOGLE_CLIENT_SECRET']
+
+  if (!clientId) {
+    throw new Error("Missing GOOGLE_CLIENT_ID environment variable.")
+  }
+
+  if (!clientSecret) {
+    throw new Error("Missing GOOGLE_CLIENT_SECRET environment variable.")
+  }
+
+  return { clientId, clientSecret, scope: ['email', 'profile'] }
+}
+
+// Default implementation if there is no `auth.methods.google.onSignInFn`.
+export async function onSignInFn(_context, args) {
+  const email = args.profile.emails[0].value
```

Comment:
Also, no optional chaining here? Is this because we expect to always end up on the happy path?

Reply:
I'd probably argue the same here against optional chaining. If the info we get from Google is promised to exist, blowing up is probably ok since there is no sensible recovery mechanism. Once we merge `UsernameAndPassword` we won't be pulling email anymore, but instead some google identifier. But again, if they say it exists as some property on an object and it is missing, we probably just want to fail fast. However, when I do make the change, I can try to do optional chaining with a better error message at least and see how that looks/feels.